src/stash-debug/src/main.rs

```diff
@@ -421,6 +422,7 @@ impl Usage {
             system_parameter_frontend: None,
             storage_usage_retention_period: None,
             connection_context: None,
+            active_connection_count: Arc::new(Mutex::new(ConnectionCounter::new(0))),
```

Comment:
I'm genuinely not opposed to just having this be an Option to avoid plumbing this business through all the tests. Thoughts?

Reply:
Yeah, I had it as an Option and changed it to an Arc... my main reason was that all the production use cases need it. (I initially set it to None for the internal port but realized that that's how the user changes the max_connections so we still need to thread it through, we just don't need to check the user). I think they're both a little awkward; I'm happy to do either